src/lib/libcmd/cut.c

```diff
@@ -258,6 +258,7 @@ static void cutcols(Cut_t *cut, Sfio_t *fdin, Sfio_t *fdout) {
         if (!bp && !(bp = sfgetr(fdin, 0, SF_LASTR))) break;
         len = sfvalue(fdin);
         xx = 0;
+        lp = cut->list;
```

Comment:
OMFG! Do you see why this initialization has to be here rather than at the top of the function? It's because of the two increments to `lp` within this module. The second is unconditional and not needed as the statement could be rewritten to avoid the otherwise gratuitous increment. The first, conditional, increment is because the first element of `cut->list` has a special meaning; i.e., it's not really part of the list of columns.